packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -1286,6 +1286,16 @@ function renderRoot(root: FiberRoot, isYieldy: boolean): void {
           // Record the time spent rendering before an error was thrown.
           // This avoids inaccurate Profiler durations in the case of a suspended render.
           stopProfilerTimerIfRunningAndRecordDelta(nextUnitOfWork, true);
+
+          // HACK Also propagate actualDuration for the time spent in the fiber that errored.
+          // This avoids inaccurate Profiler durations in the case of a suspended render.
+          // This happens automatically for sync renders, because of resetChildExpirationTime.
```

Comment:
Suggested alternative: bubble this in `completeUnitOfWork`, where `resetChildExpirationTime` would normally go for a completed fiber.

Reply:
We [already have a similar bubbling there](https://github.com/facebook/react/blob/8b87ebf5b0053838fb58e4ad05c24e134030d75a/packages/react-reconciler/src/ReactFiberScheduler.js#L1059-L1070) to ensure that `actualDuration` bubbles all the way up to the `Suspense` component (or error boundary).

I think it's tricky to propagate this specific value (from the fiber that threw) in `completeUnitOfWork` without counting it too many times (since we hit that code path for ancestors too).

Reply:
Scratch that, I misunderstood your initial suggestion. You're right!